Lib/email/_header_value_parser.py

```diff
@@ -566,12 +566,12 @@ def display_name(self):
         if res[0].token_type == 'cfws':
             res.pop(0)
         else:
-            if res[0][0].token_type == 'cfws':
+            if isinstance(res[0][0], TokenList) and res[0][0].token_type == 'cfws':
```

Comment:
You only need to check `res[0]` is a `TokenList` and not `rest[0][0]`. Same is true for the fix on line 574.